log-output/pkg/index/index.go

```diff
@@ -0,0 +1,36 @@
+package index
+
+import "time"
+
+const MaxIndexLogs = 100
+
+type Index struct {
+	Name     string
+	FirstLog time.Time `json:"first_log"`
+	LastLog  time.Time `json:"last_log"`
+	LogCount int64     `json:"log_count"`
+	Logs     [][]byte  `json:"-"`
+}
+
+func NewIndex(name string) Index {
+	return Index{
+		Name: name,
+		Logs: make([][]byte, MaxIndexLogs),
+	}
+}
+
+func WriteLog(index *Index, _log []byte) {
+	if index.FirstLog.IsZero() {
+		index.FirstLog = time.Now()
+	}
+	index.LastLog = time.Now()
+
+	index.LogCount++
+
+	if index.LogCount > MaxIndexLogs {
+		index.Logs = append(index.Logs[1:], _log)
+	} else {
+		index.Logs[index.LogCount-1] = _log
```

Comment:
Should the index increase here?

Reply:
The index is a fixed size because it is stored in memory.  The intent is to give a sample of the logs received and track how many logs were written, not store all the logs.